### PR TITLE
split sbom (icm.json) to arch specific files

### PR DESCRIPTION
**Pull request by rcerven**
* STONEBLD-1158

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] Python type annotations added to new code
- [n/a] JSON/YAML configuration changes are updated in the relevant schema
- [n/a] Changes to metadata also update the documentation for the metadata
- [n/a] Pull request has a link to an osbs-docs PR for user documentation updates
- [n/a] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/generate_sbom.py [hunk 1]
```diff
@@ -181,16 +183,26 @@ def get_sbom_json(self, sbom_url: str) -> Dict[str, Any]:
...
 
         archives = self.koji_session.listArchives(build['build_id'], type=KOJI_BTYPE_ICM)
         sbom_path = self.pathinfo.typedir(build, btype=KOJI_BTYPE_ICM)
         for archive in archives:
-            if archive['filename'] == ICM_JSON_FILENAME:
-                sbom_url = os.path.join(sbom_path, ICM_JSON_FILENAME)
-                break
-        return sbom_url
+            for platform in self.all_platforms:
+                arch_icm_filename = ICM_JSON_FILENAME.format(platform)
```
Comment (mkosiarc) [inline atomic_reactor/plugins/generate_sbom.py:193]: this nested for loop does not seem right, we should not regenerate this for every archive
I prefer something like:
```
...
icm_filenames_platforms = {ICM_JSON_FILENAME.format(platform): platform for platform in self.all_platforms}
for archive in archives:
    if archive['filename'] in icm_filenames_arches:
        platform = icm_filenames_arches[archive['filename']]
        sbom_urls[platform] = os.path.join(sbom_path, archive['filename'])
...   
```
This way we could avoid nested loop and a break statement. What do you think?

Comment (rcerven) [reply atomic_reactor/plugins/generate_sbom.py]: changed

File: atomic_reactor/plugins/generate_sbom.py [hunk 2]
```diff
@@ -203,47 +214,68 @@ def get_parent_images_nvr(self) -> List[Optional[str]]:
...
+            sbom_urls = self.get_sbom_urls_from_build(build)
+
+            if sbom_urls:
+                for platform in self.all_platforms:
+                    parent_image_sbom_json = self.get_sbom_json(sbom_urls[platform])
+                    parent_components[platform] = parent_image_sbom_json['components']
+
+                    # add reasons from parent images
+                    if 'incompleteness_reasons' in parent_image_sbom_json:
+                        for reason in parent_image_sbom_json['incompleteness_reasons']:
```
Comment (mkosiarc) [inline atomic_reactor/plugins/generate_sbom.py:236]: Maybe
```
for reason in parent_image_sbom_json.get('incompleteness_reasons', {}):
```
would improve the readability, as we could omit the condition and one nested level

**Comment by mkosiarc** [comment]
LGTM with one minor comment. Seems like the pipeline is broken, even in master. I'll try to see what's going on.

File: atomic_reactor/plugins/generate_sbom.py (hunk 2 above)
Comment (rcerven) [reply atomic_reactor/plugins/generate_sbom.py]: changed

**Review by mkosiarc — APPROVED** [review]

**Review by ben-alkov — APPROVED** [review]

**Comment by rcerven** [comment]
rn updated

**Pull request merged**